base/essentials.jl

```diff
@@ -601,6 +601,20 @@ end
 
 Val(x) = (@_pure_meta; Val{x}())
 
+eltype(::Val{T}) where T = T
```

Comment:
This isn't correct. `T` can be `Array{S,N}`, in which case `eltype(Val{Array{S,N}})` should be `S`. I'd instead make this `eltype(::Val{T}) where {T} = eltype(T)`.

Reply:
Hmm maybe eltype isn't the right name for what I want. I just want a function that will turn `Val{:a}()` into `:a`

Reply:
However, might be consistent with this:
```
Julia-0.7.0-DEV> x = Vector{Array{Int,3}}()
0-element Array{Array{Int64,3},1}

Julia-0.7.0-DEV> eltype(x)
Array{Int64,3}
```

Reply:
`Val` isn't a collection or the type of a collection so `eltype` is not appropriate here.

Reply:
I mean I could just leave it out this PR, or choose another name? `devalue`?